src/main/webapp/src/components/songInfo.js

```diff
@@ -94,18 +100,53 @@ class SongInfo extends React.Component {
     });
   }
 
+  /** Save state of the component to localStorage. */
+  saveState() {
+    const json = JSON.stringify(this.state);
+    localStorage.setItem("state", json);
+  }
+
+  /** Checks if the state of the component was loaded successfully. */
+  isReady(componentState) {
+      return (componentState && !componentState.isLoading && !componentState.errorMsg);
+  }
+
+  /** If all analysis components are ready, send SongInfo object to the server. */
+  sendSongInfo(state) {
+    if (!state.wasSent && this.isReady(state.youTubeState) && this.isReady(state.entityState) && this.isReady(state.sentimentState)) {
```

Comment:
It would probably be better to extract `this.isReady(state.youTubeState) && this.isReady(state.entityState) && this.isReady(state.sentimentState)` to a separate function, e.g. `isReady(state)`, so the if condition is better to read.